avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroDeserializer.java

```diff
@@ -71,4 +78,28 @@ protected Object deserialize(byte[] payload) throws SerializationException {
       throw new SerializationException("Error deserializing Avro message for id " + id, e);
     }
   }
+
+    private DatumReader getDatumReader(Schema schema) {
+        if (useSpecificAvroReader) {
+            return new SpecificDatumReader(schema, getTargetSchema(schema));
+        } else {
+            return new GenericDatumReader(schema);
+        }
+    }
+
+    private Schema getTargetSchema(Schema schema) {
+        Schema targetSchema = targetSchemaCache.get(schema);
+        if (targetSchema == null) {
+            Class targetClass = SpecificData.get().getClass(schema);
+            try {
+                targetSchema = ((SpecificRecord)targetClass.newInstance()).getSchema();
+            } catch (InstantiationException e) {
+                throw new RuntimeException("Failed to get target schema for " + targetClass, e);
```

Comment:
Since this is just going to be converted to a SerializationException when it gets caught in `deserialize()`, maybe we could just throw a SerializationException directly here?

We might also want to make the error messages a bit clearer here, maybe saying that a SpecificRecord was requested by the class specified by the schema could not be found.